src/redisearch_rs/inverted_index/src/full.rs

```diff
@@ -114,10 +110,25 @@ fn decode_term_record_offsets<'a>(
 
     cursor.set_position(end as u64);
 
-    record.data.term = ManuallyDrop::new(RSTermRecord {
+    let term_record = RSTermRecord {
         term: std::ptr::null_mut(),
         offsets: RSOffsetVector::with_data(data, offsets_sz),
-    });
+    };
+
+    let record = RSIndexResult {
+        data: RSIndexResultData {
+            term: ManuallyDrop::new(term_record),
+        },
+        result_type: RSResultType::Term,
+        doc_id: base + delta as t_docId,
+        field_mask,
+        freq,
+        offsets_sz,
+        is_copy: false,
+        dmd: std::ptr::null(),
+        metrics: std::ptr::null_mut(),
+        weight: 0.0,
+    };
```

Comment:
Isn't there a `RSIndexResult` constructor we can use instead? Maybe worth adding one?

Reply:
I extended `term_with_term_ptr()` for this.

I also added a small optimization suggested by Luca. See the two last commits.

Reply:
Did you forget to updatet this call? :see_no_evil:

Reply:
oups...